examples/basic/packages/eslint-config/package.json

```
@@ -1,19 +1,25 @@
 {
   "name": "@repo/eslint-config",
   "version": "0.0.0",
+  "type": "module",
   "private": true,
-  "files": [
-    "library.js",
-    "next.js",
-    "react-internal.js"
-  ],
+  "exports": {
+    "./base": "./base.js",
+    "./next-js": "./next.js",
+    "./react-internal": "./react-internal.js"
+  },
   "devDependencies": {
-    "@vercel/style-guide": "^5.2.0",
-    "eslint-config-turbo": "^2.0.0",
+    "@next/eslint-plugin-next": "^15",
+    "@typescript-eslint/eslint-plugin": "^7.1.0",
+    "@typescript-eslint/parser": "^7.1.0",
+    "eslint": "^9.15.0",
```

```
Issues with peer dependencies found
packages/eslint-config
├─┬ @typescript-eslint/eslint-plugin 7.18.0
│ ├── ✕ unmet peer eslint@^8.56.0: found 9.15.0
│ └─┬ @typescript-eslint/type-utils 7.18.0
│   ├── ✕ unmet peer eslint@^8.56.0: found 9.15.0
│   └─┬ @typescript-eslint/utils 7.18.0
│     └── ✕ unmet peer eslint@^8.56.0: found 9.15.0
└─┬ @typescript-eslint/parser 7.18.0
  └── ✕ unmet peer eslint@^8.56.0: found 9.15.0
```

running `pnpm update -i --latest` shows

```
pnpm update -i --latest
? Choose which packages to update (Press <space> to select, <a> to toggle all, <i> to invert selection) … 
❯ ○ devDependencies
    ○ Package                                                    Current   Target            
    ○ @typescript-eslint/eslint-plugin                            7.18.0 ❯ 8.15.0      
    ○ @typescript-eslint/parser                                   7.18.0 ❯ 8.15.0 
```

```suggestion
    "@typescript-eslint/eslint-plugin": "^8.15.0",
    "@typescript-eslint/parser": "^8.15.0",
    "eslint": "^9.15.0",
```

also, why not put `eslint` as a devDependency in the root package.json so it can be removed here and in ui pkg?

---

Oh, thanks, good catch!

Best practices on why not to put dependencies in the root are [here](https://turbo.build/repo/docs/crafting-your-repository/managing-dependencies#best-practices-for-dependency-installation). 

(or in [video form, shameless self-promo](https://www.youtube.com/watch?v=oHag57_zRs8))

---

I also ran into weird errors such as this one while having vitest at the root package.json - in case anyone else needs more persuasion

https://github.com/vitest-dev/vitest/issues/3296